src/lib/components/search-bar/search.ts

```diff
@@ -1,149 +1,112 @@
-import fuzzysort from 'fuzzysort';
-
-import tokens from '$lib/stores/tokens';
-import type { TokenInfoWrapper } from '$lib/stores/tokens/tokens.store';
-import { get } from 'svelte/store';
 import { isValidGitUrl } from '$lib/utils/is-valid-git-url';
-import GitProjectService from '$lib/utils/project/GitProjectService';
-import { isAddress } from 'ethers';
+import { MeiliSearch, type FederatedMultiSearchParams } from 'meilisearch';
+import { resultsSchema, type Result } from './types';
+import { z } from 'zod';
+import ensStore from '$lib/stores/ens/ens.store';
+import mapFilterUndefined from '$lib/utils/map-filter-undefined';
+import { BASE_URL } from '$lib/utils/base-url';
 import network from '$lib/stores/wallet/network';
+import { isAddress } from 'ethers';
 
-export enum SearchItemType {
-  PROFILE,
-  TOKEN,
-  REPO,
-}
+const client = new MeiliSearch({
+  host: `${BASE_URL}/api/search`,
+});
 
-interface MatchStrings {
-  primary?: string;
-  secondary?: string;
-  tertiary?: string;
-}
+/**
+ * Add a "fake" search result for an unclaimed project if a Git URL or owner/name pattern is entered
+ * for a repo not found in Drips DB
+ */
+async function getGitUrlResults(q: string, parsedHits: Result[]): Promise<Result | undefined> {
+  if (!isValidGitUrl(q)) return undefined;
 
-export type Item =
-  | {
-      type: SearchItemType.PROFILE;
-      matchStrings: MatchStrings;
-      item: {
-        address?: string;
-        name?: string;
-        dripsAccountId?: string;
-      };
-    }
-  | {
-      type: SearchItemType.TOKEN;
-      matchStrings: MatchStrings;
-      item: TokenInfoWrapper;
-    }
-  | {
-      type: SearchItemType.REPO;
-      matchStrings: MatchStrings;
-      item: {
-        forge: string;
-        repoName: string;
-        username: string;
-        url: string;
-      };
-    };
+  const gitHubUrl = isValidGitUrl(q) ? q : `https://github.com/${q}`;
```

Comment:
`isValidGitUrl` is probably not expensive, but you might save the return value from above for the check here anyways

Reply:
ah that second check is completely unnecessary actually